tests/Aspire.Hosting.Azure.Tests/AzureFunctionsTests.cs

```diff
@@ -94,6 +96,39 @@ public void AddAzureFunctionsProject_WiresUpHttpEndpointCorrectly_WhenPortArgume
         Assert.True(endpointAnnotation.IsProxied);
     }
 
+    [Fact]
+    public void AddAzureFunctionsProject_GeneratesUniqueDefaultHostStorageResourceName()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+        builder.AddAzureFunctionsProject<TestProjectWithMalformedPort>("funcapp");
+
+        // Assert that the default storage resource is unique
+        var storageResources = Assert.Single(builder.Resources.OfType<AzureStorageResource>());
+        Assert.NotEqual(AzureFunctionsProjectResourceExtensions.DefaultAzureFunctionsHostStorageName, storageResources.Name);
+        Assert.StartsWith(AzureFunctionsProjectResourceExtensions.DefaultAzureFunctionsHostStorageName, storageResources.Name);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task AddAzureFunctionsProject_RemoveDefaultHostStorageWhenUseHostStorageIsUsed()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+        var storage = builder.AddAzureStorage("my-own-storage").RunAsEmulator();
+        builder.AddAzureFunctionsProject<TestProjectWithMalformedPort>("funcapp")
+            .WithHostStorage(storage);
+
+        using var host = builder.Build();
+        await host.StartAsync();
+
+        // Assert that the default storage resource is not present
+        var model = host.Services.GetRequiredService<DistributedApplicationModel>();
+        Assert.DoesNotContain(model.Resources.OfType<AzureStorageResource>(),
+            r => r.Name.StartsWith(AzureFunctionsProjectResourceExtensions.DefaultAzureFunctionsHostStorageName));
+        Assert.Contains(model.Resources.OfType<AzureStorageResource>(), r => r.Name == "my-own-storage");
```

Comment:
Can we make this test even stronger and assert there is only 1 storage resource?